### PR TITLE
Added preview of git logs to start of tag-release process

### PR DESCRIPTION
**Pull request by andrialexandrou**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "test": "BABEL_ENV=test ava 'test/**/*.js' --require ./test/helpers/setup.js",
     "test:nyan": "npm test -- --tap | tap-nyan",
     "test:watch": "npm test -- --watch",
-    "cover": "nyc --reporter=text --reporter=html --check-coverage --statements 97.56 --branches 95.92 --functions 98.39 --lines 97.56 npm test",
+    "cover": "nyc --reporter=text --reporter=html --check-coverage --statements 90 --branches 90 --functions 90 --lines 90 npm test",
```
Comment (elijahmanor) [inline package.json]: While we are changing this, let's only check statements here (so please remove branches, functions, and lines) since Jim confirmed to me the other day that is the only one we are enforcing. Of course we can shoot for higher, but 90% statements should be the low bar.

File: src/index.js [hunk 2]
```diff
@@ -79,14 +80,21 @@ export function startTagRelease( options, queries = questions.general ) {
...
+	], options )
 		.catch( error => {
 			utils.prompt( questions.github ).then( answers => {
 				const { username, password } = answers;
 				utils.createGitHubAuthToken( username, password ).then( token => {
 					utils.setGitConfigs( username, token );
-					startTagRelease( { username, token } );
+					options.username = username;
+					options.token = token;
+					startTagRelease( options );
```
Comment (elijahmanor) [inline src/index.js]: Why was this changed from `startTagRelease( { username, token } );` to this? Is something getting added to `options` that I'm missing?

File: src/sequence-steps.js [hunk 3]
```diff
@@ -279,7 +248,10 @@ export function gitPushOriginMaster( [ git, options ] ) {
 export function githubUpstream( [ git, options ] ) {
 	const command = `git config remote.upstream.url`;
 	return utils.exec( command ).then( data => {
-		const [ , owner, name ] = data.match( /github.com[:/](.*)\/(.*)\.git/ ) || [];
+		let [ , owner, name ] = data.match( /github.com[:/](.*)\/(.*)/ ) || []; // eslint-disable-line prefer-const
```
Comment (elijahmanor) [inline src/sequence-steps.js]: In what case were you getting `.git` in the name? I'd rather update the regex to fix that instead of adding another statement to filter that out. Do you have an example of a URL that was breaking the original code?

File: src/utils.js [hunk 4]
```diff
@@ -108,18 +109,61 @@ export default {
...
 		return this.exec( `${ GIT_CONFIG_COMMAND } ${ name } ${ value.trim() }` );
 	},
-	getGitConfigs() {
+	getGitConfigs( options ) {
 		return Promise.all( [
 			this.getGitConfig( "tag-release.username" ),
 			this.getGitConfig( "tag-release.token" )
-		] );
+		] )
+			.then( config => {
```
Comment (elijahmanor) [inline src/utils.js]: Another way to code this with ES6 could be...

```
.then( ( [ username, token ] ) => {
  return Object.assign( options, { username, token } );
} )
```
...it destructures the array into named parts and then creates a new object and overlays that on top of the existing options.

File: src/index.js (hunk 2 above)
Comment (andrialexandrou) [inline src/index.js]: Yes, from the pipeline above we're able to read the git log even if not authed in github, and those logs still carry over and are stored in `options.log`, and need to be included in the function call `startTagRelease( options )` along with username and token.

File: src/sequence-steps.js (hunk 3 above)
Comment (andrialexandrou) [inline src/sequence-steps.js]: If someone's remotes are using https rather ssh, but yr right Regex does that for us! Changing....

**Comment by elijahmanor** [comment]
Great work @andrialexandrou 

Consider this merged!

selfie-2
